decode_test.go

```diff
@@ -2604,6 +2607,38 @@ func TestUnmarshalablePtrInt(t *testing.T) {
 	})
 }
 
+func TestUnmarshalableErrors(t *testing.T) {
```

Comment:
There is only a test case for `UnmarshalYAML([]byte) error`, so please add tests for other cases as well. In the tests, ensure that you can obtain an UnmarshalerError and verify that the content of its Error() matches the expected message.